src/Sequence.cc

```diff
@@ -66,8 +67,8 @@ Sequence::Sequence(gul14::string_view label)
 
 void Sequence::set_label(gul14::string_view new_label)
 {
-    check_label(new_label);
-    label_.assign(new_label.data(), new_label.size());
+    auto checked_label = check_label(new_label);
+    label_.assign(checked_label.data(), checked_label.size());
```

Comment:
Now that we are dealing with a `std::string`, there is no real reason for the `assign()`:
```suggestion
    label_ = trim_and_check_label(new_label);
```

Reply:
Yep! Your are right. Fixed.